workspaces/diff-engine/cli/src/main.rs

```diff
@@ -24,10 +24,28 @@ fn main() {
     .arg(
       Arg::with_name("specification")
         .required(true)
-        .value_name("spec-file-path")
-        .help("Sets the specification file that describes the API spec")
+        .value_name("SPEC_PATH")
+        .help("The path to the specification that describes the API spec")
         .takes_value(true),
     )
+    .arg(
+      Arg::with_name("use-spec-file")
+        .short("f")
+        .takes_value(false)
+        .help("SPEC_PATH is expected to be a single file (default)"),
+    )
+    .arg(
+      Arg::with_name("use-spec-dir")
+        .short("d")
+        .takes_value(false)
+        .help("SPEC_PATH is expected to be a directory"),
+    )
+    .group(
+      ArgGroup::with_name("spec-type")
+        .args(&["use-spec-file", "use-spec-dir"])
+        .multiple(false)
+        .required(false),
+    )
```

Comment:
This allows both the `assembly` and `diff` commands of the CLI to specify whether the passed in spec should be expected to be a single file, or a directory that needs to be assembled. Putting them together in an `ArgGroup` allows to make sure they're not used in a conflictingly.